### PR TITLE
Fix cloning of datetimes

### LINKED ISSUE #38: Notice: Undefined property: DateTime::$date
Since I updated to PHP 7.0.7 I have an issue duplicating objects with a DateTime property. Every version before 7.0.7 worked without any issues.

the error I am getting is;
`Notice: Undefined property: DateTime::$date`

The error is thrown at this line:
`$propertyValue = $property->getValue($object);`
https://github.com/myclabs/DeepCopy/blob/a8773992b362b58498eed24bf85005f363c34771/src/DeepCopy/DeepCopy.php#L179

there are older bug reports from PHP regarding this issue, so I am unsure if this library should be improved or that this is just PHP related;
http://stackoverflow.com/questions/14084222/why-cant-i-access-datetime-date-in-phps-datetime-class-is-it-a-bug

Also see my try on 3v4l to isolate the issue;
https://3v4l.org/NOGDr

### PR DESCRIPTION
**Pull request by theofidry**
Fix #38 and comes as a replacement of #42. While it is possible that some cases are missed, this is an easy fix for now. IMO, as long as we don't have a clearer view on the other edge cases, it is better to try to stay simple.

### CODE REVIEW

File: src/DeepCopy/DeepCopy.php [hunk 1]
```diff
@@ -141,6 +141,9 @@ private function copyObject($object)
         $newObject = clone $object;
         $this->hashMap[$objectHash] = $newObject;
 
+        if ($newObject instanceof \DateTimeInterface) {
+            return $newObject;
```
Comment (Ocramius) [inline src/DeepCopy/DeepCopy.php]: Not a copy

File: tests/DeepCopyTest/DeepCopyTest.php [hunk 2]
```diff
@@ -42,6 +42,17 @@ public function testPropertyObjectCopy()
...
 
+    public function testPropertyObjectCopyWithDateTimes()
+    {
+        $o = new A();
+        $o->date1 = new \DateTime();
+        $o->date2 = new \DateTimeImmutable();
+
+        $deepCopy = new DeepCopy();
+
+        $this->assertDeepCopyOf($o, $deepCopy->copy($o));
```
Comment (Ocramius) [inline tests/DeepCopyTest/DeepCopyTest.php]: Write a test that mutates `$o->date1` and assert that the data is different in the deep copy.

File: src/DeepCopy/DeepCopy.php (hunk 1 above)
Comment (theofidry) [inline src/DeepCopy/DeepCopy.php]: it is from L141

**Comment by theofidry** [comment]
Thanks for the review @Ocramius, everything is green

**Comment by theofidry** [comment]
@peternijssen @mnapoli do you mind taking a look quickly? I don't like to be pressing, but this issue affects anyone using PHP7.0.7+ and downgrading is not always an option.

**Comment by peternijssen** [comment]
I am in for any fix. Currently I am using my fix in production with quiet a large application and it runs fine, but as I noted before I had no clue what we should do with the tests.

I prefer your fix for now since it's covered by tests.

File: src/DeepCopy/DeepCopy.php (hunk 1 above)
Comment (Ocramius) [inline src/DeepCopy/DeepCopy.php]: Oh, never mind :-)

**Comment by theofidry** [comment]
ping @mnapoli

**Comment by mnapoli** [comment]
Very sorry about the delay, I'm looking at it right now :)

**Comment by theofidry** [comment]
Cheers :)

**Comment by mnapoli** [comment]
Let's ship this! Thanks @theofidry and @peternijssen for your work

**Pull request merged**